### PR TITLE
docs: reorganise release/version docs

### LINKED ISSUE #7031: Repo: Why the version bump to `v5.59.6`?
### Suggestion

Looking at the [changes between `v5.59.5` and `v5.59.6`](https://github.com/typescript-eslint/typescript-eslint/compare/v5.59.5...v5.59.6), there was only a single commit (for the version bump), none of the changelogs updated in that commit indicate what changed and the only packages updated were `@typescript-eslint` scoped packages.

I assume that one of the updated dependencies has its own repo and changelog, but there doesn't seem to be any way to know what triggered the version bump other than by going to each typescript-eslint repo and checking. Would it be possible to add this information to the changelog?

### PR DESCRIPTION
**Pull request by bradzacher**
## PR Checklist

- [x] Addresses an existing open issue: fixes #7031

## Overview

Slight cleanup / reorganisation / deduplication of the docs here

To start with - the new sidebar:
<p align="center">

</p>

> Note that "Contributing" and "Maintenance" are collapsed by default.

Sidebar Changes:
- "Users" section to clearly group docs made for users
  - We now have these clearly user-facing docs where previously it was hidden inside the "maintenance" docs and looked like it was internal docs:
    - "Dependency Versions" explains what dependency versions we support
    - "Releases" explains when and why we do releases
    - "Versioning" explains how we classify our version bumps
- "Developers" section to clearly group docs made for developers
  - For now just contains "Building Custom Rules" but can house more in future if we write some.
- Renamed "Architecture" to "Packages". The former didn't really describe what it contained to me, and IMO it was super hard to find the package documentation unless you go directly via the npm package page.

You can follow the changes but I moved some sections around amongst the "User" docs to deduplicate. EG I noticed there wasn't really clear delineation between the "releases" doc and the "versioning" doc - and there was some duplication between them too.

---

@JamesHenry - is there any way to add a custom footer to the GH release notes?
I'd love it if we could add a link to both the "Releases" and "Versioning" pages to the release notes so that we can very clearly promote these docs.

### CODE REVIEW

File: docs/Packages.mdx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+id: packages
+title: Packages
+---
+
+The `@typescript-eslint/*` packages are built from a monorepo located at https://github.com/typescript-eslint/typescript-eslint.
```
Comment (Josh-Cena) [inline docs/Packages.mdx:6]:
```suggestion
The `@typescript-eslint/*` packages are all stored in [our GitHub monorepo](https://github.com/typescript-eslint/typescript-eslint).
```

File: docs/Packages.mdx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+The monorepo is built with [Lerna](https://lerna.js.org) and [Nx](https://nx.dev).
+
+Each page in this section corresponds to a package we intentionally expose to users.
+They are:
+
+- [`@typescript-eslint/eslint-plugin`](./packages/ESLint_Plugin.mdx): An ESLint plugin which provides lint rules for TypeScript codebases.
+- [`@typescript-eslint/eslint-plugin-tslint`](./packages/ESLint_Plugin_TSLint.mdx): ESLint plugin that allows running TSLint rules within ESLint to help you migrate from TSLint to ESLint.
+- [`@typescript-eslint/parser`](./packages/Parser.mdx): An ESLint parser which allows for ESLint to lint TypeScript source code.
+- [`@typescript-eslint/scope-manager`](./packages/Scope_Manager.mdx): A fork of [`eslint-scope`](https://github.com/eslint/eslint-scope), enhanced to support TypeScript functionality.
+- [`@typescript-eslint/typescript-estree`](./packages/TypeScript-ESTree.mdx): The underlying code used by [`@typescript-eslint/parser`](./packages/Parser.mdx) that converts TypeScript source code into an <a href="https://github.com/estree/estree">ESTree</a>-compatible form.
```
Comment (Josh-Cena) [inline docs/Packages.mdx:16]: Does Markdown link syntax not work here?

File: docs/Packages.mdx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+---
+
+The `@typescript-eslint/*` packages are built from a monorepo located at https://github.com/typescript-eslint/typescript-eslint.
+The monorepo is built with [Lerna](https://lerna.js.org) and [Nx](https://nx.dev).
+
+Each page in this section corresponds to a package we intentionally expose to users.
+They are:
+
+- [`@typescript-eslint/eslint-plugin`](./packages/ESLint_Plugin.mdx): An ESLint plugin which provides lint rules for TypeScript codebases.
+- [`@typescript-eslint/eslint-plugin-tslint`](./packages/ESLint_Plugin_TSLint.mdx): ESLint plugin that allows running TSLint rules within ESLint to help you migrate from TSLint to ESLint.
```
Comment (Josh-Cena) [inline docs/Packages.mdx:13]:
```suggestion
- [`@typescript-eslint/eslint-plugin-tslint`](./packages/ESLint_Plugin_TSLint.mdx): An ESLint plugin that allows running TSLint rules within ESLint to help you migrate from TSLint to ESLint.
```

File: docs/Users.mdx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+---
+id: users
+title: Users
+---
+
+These are the user guides to consuming the typescript-eslint tooling.
+It's intended for users who want to consume our tools.
+
+> If you're reading this as a new user: welcome to the community!
+> We're happy to have you! ❤️‍🔥
```
Comment (Josh-Cena) [inline docs/Users.mdx:10]: I suggest not using block quotes for callouts.

```suggestion
:::info Welcome!
If you're reading this as a new user: welcome to the community!
We're happy to have you! ❤️‍🔥
:::
```

**Review by Josh-Cena — COMMENTED** [review]
Incomplete reviews. Can we also settle on a file naming convention? We seem to be using more kebab-casing (and some SCREAM_CASING) in the past.

File: docs/Packages.mdx (hunk 2 above)
Comment (bradzacher) [reply docs/Packages.mdx:16]: 🤷 copy 🍝 I have no idea why this uses an `<a>`

File: docs/Users.mdx (hunk 4 above)
Comment (bradzacher) [reply docs/Users.mdx:10]: Copy 🍝
Happy to improve it!

**Comment by bradzacher** [comment]
Build error

```
3:04:51 PM: [ERROR] Unable to build website for locale en.
3:04:51 PM: [ERROR] Error: Docusaurus found broken links!
3:04:51 PM: 
3:04:51 PM: Please check the pages of your site in the list below, and make sure you don't reference any path that does not exist.
Note: it's possible to ignore broken links with the 'onBrokenLinks' Docusaurus configuration, and let the build pass.
3:04:51 PM: Exhaustive list of all broken links found:
3:04:51 PM: 
3:04:51 PM: - On source page path = /packages/utils:
3:04:51 PM:    -> linking to /custom-rules
```

Weirdly it didn't show up locally but 🤷‍♂️

File: packages/website/sidebars/sidebar.base.js [hunk 5]
```diff
@@ -15,40 +17,64 @@ module.exports = {
...
     },
-    'custom-rules',
     {
+      collapsible: false,
+      items: [
+        'users/dependency-versions',
+        'users/releases',
+        'users/versioning',
+      ],
+      label: 'Users',
```
Comment (JoshuaKGoldberg) [inline packages/website/sidebars/sidebar.base.js:48]: I think this is the only part of this PR I'm not 100% positive on... Everybody is a user. Is there a more precise term we can use? _End Users_ perhaps?

**Review by JoshuaKGoldberg — DISMISSED** [review]
Cartoon penguin happily saying 'primo!'

**Comment by Josh-Cena** [comment]
> there were a number of times that I moved files around and then ran `yarn start` and it wouldn't detect broken links (where we used relative file links to other mdx files). Is this expected with `yarn start` and it's only guaranteed to be detected with a proper build?

Yes, the broken link detection only works reliably in build. Sometimes dev also reports link errors but I can't remember in which cases (usually when it blocks rendering the correct link)

File: packages/website/sidebars/sidebar.base.js (hunk 5 above)
Comment (bradzacher) [reply packages/website/sidebars/sidebar.base.js:48]: My thinking was there are "users" which are the ones doing linting and the "developers" which are the ones building things with their tooling.

Technically the "developers" are "users" of our tooling, for sure, but I thought there might be enough distinction?

I don't love "end users"... I wonder if there's another term we can use?

**Pull request merged**